tasks/task_manager.cc

```diff
@@ -379,7 +390,31 @@ task_manager::virtual_task::impl::impl(module_ptr module) noexcept
 {}
 
 future<std::vector<task_identity>> task_manager::virtual_task::impl::get_children(module_ptr module, task_id parent_id) {
-    return make_ready_future<std::vector<task_identity>>();
+    auto ms = module->get_task_manager()._messaging;
+    if (!ms) {
+        auto ids = co_await module->get_task_manager().get_virtual_task_children(parent_id);
+        co_return boost::copy_range<std::vector<task_identity>>(ids | boost::adaptors::transformed([&tm = module->get_task_manager()] (auto id) {
+            return task_identity{
+                .node = tm.get_broadcast_address(),
+                .task_id = id
+            };
+        }));
+    }
+
+    auto nodes = module->get_nodes();
+    co_return co_await map_reduce(nodes, [ms, parent_id] (auto addr) -> future<std::vector<task_identity>> {
+        return ms->send_tasks_get_children(netw::msg_addr{addr}, parent_id).then_wrapped([addr] (auto resp) {
+            if (resp.failed()) {
+                return std::vector<task_identity>{};
```

Comment:
A user doesn't choose the list of nodes that need to be connected to in order to gather children info. We use nodes returned by `module::get_nodes`. If a node to which we try to connect is down (and we don't know it yet), then I believe we should not raise.

Maybe we should log the error here and also in tasks_get_children_handler (in case it's not communication error) so that the failure could be examined.

WDYT?

Reply:
I think it is better to fail the operation. What I'm afraid of is that this can hide bugs. It is hard or next to impossible to tell apart a connection error from some other error here, so we ignore all. This can cover up bugs.
If throwing here becomes problematic, we can revisit this in the future.